src/gfnff/gfnff_setup.f90

```diff
@@ -239,5 +244,116 @@ subroutine write_gfnff_adjacency(fname, topo)
 
 end subroutine write_gfnff_adjacency
 
+! 
+subroutine wrtopo(iwrtopo,topo,n)
+  integer, intent(in) :: iwrtopo(21)
+  type(TGFFTopology), intent(inout) :: topo
+  integer :: n ! num atoms
+  !!! list to print
+  ! num lists to print
+  integer :: nprint
+  integer :: i,j,k
+  integer :: lin
+  logical :: f_ex
+  nprint=0
+  do i=1, 21
+    if(iwrtopo(i).ne.0) nprint=nprint+1
+  enddo
+
+  do i=1, nprint
+  if(iwrtopo(i).eq.1)then ! nb(20,n) 
+     INQUIRE(FILE='topo_nb', EXIST=f_ex)
+     if(f_ex)then
+       open(43,file='topo_nb')
+       close(43,status='delete')
```

Comment:
Let's use our `open_file` and `close_file` routines for this purpose. Also, we don't use hardcoded file units in `xtb`.